deepchem/feat/base_classes.py

```diff
@@ -207,6 +210,198 @@ def __call__(self, molecules):
     return self.featurize(molecules)
 
 
+class StructureFeaturizer(Featurizer):
+  """
+  Abstract class for calculating a set of features for an
+  inorganic crystal structure.
+
+  The defining feature of a `StructureFeaturizer` is that it
+  operates on 3D crystal structures with periodic boundary conditions. 
+  Inorganic crystal structures are represented by Pymatgen structure
+  objects. Featurizers for inorganic crystal structures that are subclasses of
+  this class should plan to process input which comes as pymatgen
+  structure objects. 
+
+  This class is abstract and cannot be invoked directly. You'll
+  likely only interact with this class if you're a developer. Child 
+  classes need to implement the _featurize method for calculating 
+  features for a single crystal structure.
+
+  Notes
+  -----
+  Some subclasses of this class will require pymatgen and matminer to be
+  installed.
+
+  """
+
+  def featurize(self, structures: Iterable[JSON],
+                log_every_n: int = 1000) -> np.ndarray:
+    """Calculate features for crystal structures.
+
+    Parameters
+    ----------
+    structures: Iterable[JSON]
+      Iterable sequence of pymatgen structure dictionaries.
+      Json-serializable dictionary representation of pymatgen.core.structure
+      https://pymatgen.org/pymatgen.core.structure.html
```

Comment:
I find this description a bit confusing, along with the name `JSON`.  JSON is a serialization format, and this argument is not serialized.  Yes, it *could* be serialized as JSON, but it could just as easily be serialized as XML, pickle, or many other formats.  Could we make this clearer?

Reply:
Sounds good, I'll clarify this docstring.